packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -771,8 +771,22 @@ class PageTransitionsTheme with Diagnosticable {
       platform = TargetPlatform.iOS;
     }
 
-    final PageTransitionsBuilder matchingBuilder =
-      builders[platform] ?? const ZoomPageTransitionsBuilder();
+    PageTransitionsBuilder getTransitionBuilder() {
+      late PageTransitionsBuilder defaultBuilder;
+      switch (platform) {
+        case TargetPlatform.iOS:
+          defaultBuilder = const CupertinoPageTransitionsBuilder();
+        case TargetPlatform.android:
+        case TargetPlatform.fuchsia:
+        case TargetPlatform.windows:
+        case TargetPlatform.macOS:
+        case TargetPlatform.linux:
+          defaultBuilder = const ZoomPageTransitionsBuilder();
+      }
+      return builders[platform] ?? defaultBuilder;
+    }
+
+    final PageTransitionsBuilder matchingBuilder = getTransitionBuilder();
```

Comment:
With those fancy switch expressions this could probably simplify to something like:

```dart
final PageTransitionsBuilder matchingBuilder = builders[platform] ?? switch(platform) {
  TargetPlatform.iOS => const CupertinoPageTransitionsBuilder(),
  TargetPlatform.android || TargetPlatform.fuchsia || TargetPlatform.windows || TargetPlatform.macOS || TargetPlatform.linux => const ZoomPageTransitionsBuilder(),
}
```